presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -362,7 +364,8 @@ public String quoted(RemoteTableName remoteTableName)
     @Override
     public Map<String, Object> getTableProperties(ConnectorSession session, JdbcTableHandle tableHandle)
     {
-        return delegate.getTableProperties(session, tableHandle);
+        TableHandleCacheKey key = new TableHandleCacheKey(JdbcIdentity.from(session), tableHandle.getSchemaTableName());
+        return get(tableProperties, key, () -> delegate.getTableProperties(session, tableHandle));
```

Comment:
You should make a defensive & immutable copy of the map returned by the delegate, especially that you will then share the map with multiple callers.

Can the table properties map contain null values?
if, can we remove them so that we can use ImmutableMap without need to convert to & from optionals?

cc @kokosing

Reply:
Can't I rely that delegate already returned such copy?

> Can the table properties map contain null values?

I think each connector can define its own behaviour. 
Nevertheless I think it has to be resolved on delegate level and caching level should be bothered only with caching.

Reply:
> Can't I rely that delegate already returned such copy?

If delegate returned an immutable map, that's great, but we shouldn't assume that

> caching level should be bothered only with caching.

caching introduces object reuse and sharing between threads, so it comes with additional provisions

Reply:
> Can the table properties map contain null values?

I think so. Please make sure you have tests for that.